gammapy/spectrum/models.py

```diff
@@ -574,30 +577,131 @@ class TableModel(SpectralModel):
         Array with the values of the model at energies ``energy``.
     amplitude : float
         Model amplitude that is multiplied to the supplied arrays. Defaults to 1.
+    scale_logy : boolean
+        interpolation can be done linearly or in logarithm
     """
 
-    def __init__(self, energy, values, amplitude=1):
+    def __init__(self, energy, values, amplitude=1, scale_logy=True):
         from scipy.interpolate import interp1d
         self.parameters = Bunch(amplitude=amplitude)
         self.energy = energy
         self.values = values
+        self.scale_logy = scale_logy
 
         loge = np.log10(self.energy.to('eV').value)
         try:
             self.unit = self.values.unit
-            logy = np.log10(self.values.value)
+            if scale_logy is True:
+                y = np.log10(self.values.value)
+            else:
+                y = self.values.value
         except AttributeError:
             self.unit = u.Unit('')
-            logy = np.log10(self.values)
+            if scale_logy is True:
+                y = np.log10(self.values)
+            else:
+                y = self.values
+        self.interpy = interp1d(loge,
+                                y,
+                                fill_value=-np.Inf,
+                                bounds_error=False,
+                                kind='cubic')
 
-        self.interplogy = interp1d(loge,
-                                   logy,
-                                   fill_value=-np.Inf,
-                                   bounds_error=False,
-                                   kind='cubic')
+    @classmethod
+    def from_file(cls, filename, param):
+        """A Table containing aborbed values from a XSPEC model
+        as a function of energy.
+        Todo:
+        Format of the file should be described and discussed in
+        https://gamma-astro-data-formats.readthedocs.io/en/latest/index.html
+
+        Parameters
+        ----------
+        filename : `str`
+            File containing the XSPEC model
+        param : float
+            Model parameter value
+        Examples
+        --------
+        Fill table from an EBL model (Franceschini, 2008)
+            from gammapy.spectrum.models import TableModel
+            filename = '$GAMMAPY_EXTRA/datasets/ebl/ebl_franceschini.fits.gz'
+            table_model = TableModel.from_file(filename=filename, param=0.3)
+        """
+        filename = str(make_path(filename))
+
+        # Check if parameter value is in range
+        table_param = Table.read(filename, hdu='PARAMETERS')
+        param_min = table_param['MINIMUM']
+        param_max = table_param['MAXIMUM']
+        assert(param >= param_min and param <= param_max), "Parameter\
```

Comment:
A coding rule for Gammapy (and Astropy / ... most projects): Never use `assert` to check inputs, raise an exception instead.


Here:
```python
if param_min <= param <= param_max:
    raise ValueError(<message here>)
```